pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/MailboxReceiveNode.java

```diff
@@ -65,15 +69,26 @@ public MailboxReceiveNode(int planFragmentId, DataSchema dataSchema, int senderS
     _exchangeType = exchangeType;
     _partitionKeySelector = partitionKeySelector;
     if (!CollectionUtils.isEmpty(fieldCollations)) {
-      _collationKeys = new ArrayList<>(fieldCollations.size());
-      _collationDirections = new ArrayList<>(fieldCollations.size());
+      int numCollations = fieldCollations.size();
+      _collationKeys = new ArrayList<>(numCollations);
+      _collationDirections = new ArrayList<>(numCollations);
+      _collationNullDirections = new ArrayList<>(numCollations);
       for (RelFieldCollation fieldCollation : fieldCollations) {
-        _collationDirections.add(fieldCollation.getDirection());
         _collationKeys.add(new RexExpression.InputRef(fieldCollation.getFieldIndex()));
+        Direction direction = fieldCollation.getDirection();
+        Preconditions.checkArgument(direction == Direction.ASCENDING || direction == Direction.DESCENDING,
+            "Unsupported ORDER-BY direction: %s", direction);
```

Comment:
unnecessary precondition check? or this is a null check?

Reply:
There are other direction enums (`STRICTLY_ASCENDING`, `STRICTLY_DESCENDING`, `CLUSTERED`) that we don't support yet

Reply:
Ah I see. Thx.